### PR TITLE
Fixing retrying logic by introducing a max_retries param

### PR DESCRIPTION
**Pull request by Lucas-C**
This code did NOT have the expected impact: https://github.com/psf/requests/blob/v2.22.0/requests/adapters.py#L117

### CODE REVIEW

**Comment by Lucas-C** [comment]
I do not really understand the remaining failure with the `test_downgrade_plugin` test.

I'd love some help to make tjhis PR pass :)

File: jenkinsapi/jenkins.py [hunk 1]
```diff
@@ -40,12 +40,13 @@ class Jenkins(JenkinsBase):
...
     """
 
+    # pylint: disable=too-many-arguments
     def __init__(
             self, baseurl,
             username=None, password=None,
             requester=None, lazy=False,
             ssl_verify=True, cert=None,
-            timeout=10, useCrumb=False):
+            max_retries=None, timeout=10, useCrumb=False):
```
Comment (lechat) [inline jenkinsapi/jenkins.py]: If somebody uses all params in positional manner, this change will break their code. I'd rather put this new parameter as last one.

File: jenkinsapi_tests/unittests/test_requester.py [hunk 2]
```diff
@@ -362,12 +362,13 @@ def fake_post(*args, **kwargs):  # pylint: disable=unused-argument
...
 
     req = Requester('foo', 'bar')
-    with pytest.raises(JenkinsAPIException):
+    with pytest.raises(JenkinsAPIException) as error:
         req.post_and_confirm_status(
             url='http://dummy',
             params={'param': 'value'},
             data='some data'
         )
+    assert 'status=500' in str(error)
```
Comment (lechat) [inline jenkinsapi_tests/unittests/test_requester.py]: Not sure where `status=500` will come from...

File: jenkinsapi/jenkins.py (hunk 1 above)
Comment (Lucas-C) [reply jenkinsapi/jenkins.py]: Sure !

File: jenkinsapi_tests/unittests/test_requester.py (hunk 2 above)
Comment (Lucas-C) [reply jenkinsapi_tests/unittests/test_requester.py]: From here: https://github.com/pycontribs/jenkinsapi/blob/master/jenkinsapi/utils/requester.py#L182

**Comment by Lucas-C** [comment]
The build is now passing

**Comment by lechat** [comment]
Thank you!

**Pull request merged**